megamek/src/megamek/common/Hex.java

```diff
@@ -181,6 +181,12 @@ public void setExits(IHex other, int direction, boolean roadsAutoExit) {
             }
 
             cTerr.setExit(direction, cTerr.exitsTo(oTerr));
+            
+            // Water gets a special treatment: Water at the board edge
+            // (hex == null) should usually look like ocean and 
+            // therefore always gets connection to outside the board 
+            if ((cTerr.getType() == Terrains.WATER) && (other == null))
+            	cTerr.setExit(direction, true);
```

Comment:
one-liner if blocks should still get { }

Reply:
That code bit is from 4 years ago when I hadnt seen the formatting guide :)